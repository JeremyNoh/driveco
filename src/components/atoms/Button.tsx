import React from 'react';
import {
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';

interface Props {
  style?: ViewStyle;
  title: string;
  onPress: () => void;
  disabled?: boolean;
  withoutStyle?: boolean;
  styleText?: TextStyle;
}

const Button: React.FC<Props> = ({
  style,
  title,
  onPress,
  disabled = false,
  withoutStyle = false,
  styleText,
}) => {
  return (
    <TouchableOpacity
      style={[
        !withoutStyle && styles.button,
        style,
        disabled && styles.disabled,
      ]}
      onPress={onPress}
      disabled={disabled}>
      <Text style={[styles.text, styleText]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    margin: 20,
    borderColor: '#136397',
    borderWidth: 1,
    alignItems: 'center',
    padding: 10,
    textAlign: 'center',
    justifyContent: 'center',
  },
  disabled: {
    pointerEvents: 'none',
    opacity: 0.5,
    backgroundColor: '#88B0CB',
  },
  text: {
    textTransform: 'uppercase',
    fontWeight: 'bold',
    margin: 0,
    textAlign: 'center',
  },
});
