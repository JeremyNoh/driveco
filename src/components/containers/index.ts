import TryLibContainer from './TryLibContainer';

export {TryLibContainer};
