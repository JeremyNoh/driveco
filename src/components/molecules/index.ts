import NfcTag from './NfcTag';
import ChangeLanguage from './ChangeLanguage';

export {NfcTag, ChangeLanguage};
