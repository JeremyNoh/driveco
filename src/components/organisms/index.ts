import TryLibOrganism from './TryLibOrganism';

export {TryLibOrganism};
