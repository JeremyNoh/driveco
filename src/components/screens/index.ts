import TryLibScreen from './TryLibScreen';

export {TryLibScreen};
