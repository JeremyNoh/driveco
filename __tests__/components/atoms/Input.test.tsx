import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';
import {Input} from '../../../src/components/atoms';

afterEach(cleanup);

describe('Input Atom', () => {
  it('renders correctly', () => {
    render(<Input name="test" onChange={() => {}} />).toJSON();
  });
  it('Match with snapshot', () => {
    const input = render(<Input name="test" onChange={() => {}} />).toJSON();
    expect(input).toMatchSnapshot();
  });
  it('should show good label', () => {
    const title = 'Test input';

    const {getByText} = render(<Input onChange={() => {}} name={title} />);

    const foundTitle = getByText(title);

    expect(foundTitle.props.children).toEqual(title);
  });
  it('should have the good placeholder', () => {
    const name = 'Test button';
    const onChangeMock = jest.fn();

    const placeholder = 'test placeholder';
    render(
      <Input placeholder={placeholder} onChange={onChangeMock} name={name} />,
    );

    fireEvent(screen.getByPlaceholderText(placeholder), 'onChangeText', 'ab');

    expect(onChangeMock).toHaveBeenCalledWith('ab');
  });
  it('should onChangeText Work', () => {
    const name = 'Test button';
    const placeholder = 'test placeholder';
    const onChangeTextMock = jest.fn();
    const CHANGE_TEXT = 'content';

    render(
      <Input
        placeholder={placeholder}
        onChange={onChangeTextMock}
        name={name}
      />,
    );

    fireEvent.changeText(screen.getByTestId('textInputId'), CHANGE_TEXT);
  });
});
