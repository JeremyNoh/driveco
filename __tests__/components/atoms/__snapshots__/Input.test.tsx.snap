// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Atom Match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#707070",
        "borderRadius": 13,
        "borderWidth": 1,
        "display": "flex",
        "height": 40,
        "justifyContent": "flex-start",
        "marginBottom": 25,
        "marginTop": 13,
        "position": "relative",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="viewInputId"
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "color": "#707070",
          "fontSize": 14,
          "left": 20,
          "paddingHorizontal": 5,
          "paddingVertical": 0,
          "position": "absolute",
          "textAlign": "center",
          "textTransform": "uppercase",
          "top": -10,
        },
        Object {
          "backgroundColor": "#E8E8E8",
        },
      ]
    }
    testID="nameTextInputId"
  >
    test
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "border": "none",
          "borderRadius": 13,
          "height": "95%",
          "paddingLeft": 10,
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="textInputId"
    value=""
  />
</View>
`;
